so_pysm_models/get_so_model.py

```diff
@@ -0,0 +1,102 @@
+import healpy as hp
+import numpy as np
+import pysm
+import os
+from astropy.utils import data
+from pysm.common import read_map, loadtxt
+
+data_dir = os.path.join(os.path.dirname(__file__), 'so_template')
+template = lambda x: os.path.join(data_dir, x)
+DATAURL = "http://portal.nersc.gov/project/cmb/so_pysm_models_data/"
+
+def get_data_from_url(filename):
+    with data.conf.set_temp("dataurl", DATAURL), data.conf.set_temp(
+        "remote_timeout", 30):
+        map_out = data.get_pkg_data_filename(filename)
+    return map_out
+
+def models(key, nside, pixel_indices=None, mpi_comm=None, small_scale=False):
+     model = eval(key)(nside, pixel_indices=pixel_indices, mpi_comm=mpi_comm, small_scale=small_scale)
+     for m in model:
+         m['pixel_indices'] = pixel_indices # include pixel indices in the model dictionary
+         m['nside'] = nside
+     return model
+
+def SO_d0(nside, pixel_indices=None, mpi_comm=None, small_scale=False):
+    if small_scale:
+        T_map = get_data_from_url('dust_T_ns4096.fits')
+        Q_map = get_data_from_url('dust_Q_ns4096.fits')
+        U_map = get_data_from_url('dust_U_ns4096.fits')
+    else:
+        T_map = template('dust_T_ns512.fits')
+        Q_map = template('dust_Q_ns512.fits')
+        U_map = template('dust_U_ns512.fits')
+    A_I = read_map(T_map, nside, field=0, pixel_indices=pixel_indices, mpi_comm=mpi_comm)
+    return [{
+        'model': 'modified_black_body',
+        'nu_0_I': 545.,
+        'nu_0_P': 353.,
+        'A_I': A_I,
+        'A_Q': read_map(Q_map, nside, field=0, pixel_indices=pixel_indices, mpi_comm=mpi_comm),
+        'A_U': read_map(U_map, nside, field=0, pixel_indices=pixel_indices, mpi_comm=mpi_comm),
+        'spectral_index': np.ones(len(A_I)) * 1.53,
+        'temp': np.ones(len(A_I)) * 19.6,
+        'add_decorrelation': False,
+    }]
+
+def SO_s0(nside, pixel_indices=None, mpi_comm=None, small_scale=False):
+    if small_scale:
+        T_map = get_data_from_url('synch_T_ns4096.fits')
+        Q_map = get_data_from_url('synch_Q_ns4096.fits')
+        U_map = get_data_from_url('synch_U_ns4096.fits')
+    else:
+        T_map = template('synch_T_ns512.fits')
+        Q_map = template('synch_Q_ns512.fits')
+        U_map = template('synch_U_ns512.fits')
+    A_I = read_map(T_map, nside, field=0, pixel_indices=pixel_indices, mpi_comm=mpi_comm)
+    return [{
+        'model': 'power_law',
+        'nu_0_I': 0.408,
+        'nu_0_P': 23.,
+        'A_I': A_I,
+        'A_Q': read_map(Q_map, nside, field=0, pixel_indices=pixel_indices, mpi_comm=mpi_comm),
+        'A_U': read_map(U_map, nside, field=0, pixel_indices=pixel_indices, mpi_comm=mpi_comm),
+        'spectral_index': np.ones(len(A_I)) * -3.1,
+    }]
+
+def SO_f0(nside, pixel_indices=None, mpi_comm=None, small_scale=False):
```

Comment:
wouldn't it be better, if this model is based on PySM's `f1` let's say, call it `SO_f1`?

Reply:
yes, changing it!

Reply:
done